sql/row: TestCheckRowSizeRateLimitsAndTruncates failed

sql/row.TestCheckRowSizeRateLimitsAndTruncates [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21361720?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21361720?buildTab=artifacts#/) on master @ [6b52d86cd0942ce6db8193020ce627fd7d19fb7e](https://github.com/cockroachdb/cockroach/commits/6b52d86cd0942ce6db8193020ce627fd7d19fb7e):


```
=== RUN   TestCheckRowSizeRateLimitsAndTruncates
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/9db56821b86748e0fc537f3d515778f0/logTestCheckRowSizeRateLimitsAndTruncates2366498335
    test_log_scope.go:82: use -show-logs to present logs inline
    helper_test.go:186: -- test log scope end --
--- FAIL: TestCheckRowSizeRateLimitsAndTruncates (1.38s)
=== RUN   TestCheckRowSizeRateLimitsAndTruncates/primary_key_truncation_via_SQL_INSERT
    helper_test.go:176: 
        	Error Trace:	pkg/sql/row/helper_test.go:176
        	Error:      	"[]" should have 1 item(s), but has 0
        	Test:       	TestCheckRowSizeRateLimitsAndTruncates/primary_key_truncation_via_SQL_INSERT
        	Messages:   	expected exactly one large_row event for the wide INSERT
--- FAIL: TestCheckRowSizeRateLimitsAndTruncates/primary_key_truncation_via_SQL_INSERT (0.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRowSizeRateLimitsAndTruncates.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64278

--- a/pkg/sql/row/helper_test.go
+++ b/pkg/sql/row/helper_test.go
@@ -154,12 +154,14 @@ func TestCheckRowSizeRateLimitsAndTruncates(t *testing.T) {
 	t.Run("primary key truncation via SQL INSERT", func(t *testing.T) {
 		logSpy.Reset()
 		skippedLargeRows.Store(0)
-		// Re-init the global limiter so the SQL INSERT below sees an open
-		// window regardless of what the previous subtest did.
-		largeRowLogEvery = log.Every(testInterval)
+		// Disable rate-limiting entirely for this subtest. The limiter is
+		// process-global, so a background internal SQL operation writing a
+		// row over the lowered 1 KiB threshold could otherwise consume the
+		// window before our INSERT runs, suppressing the event we assert on.
+		// This subtest only exercises truncation, not rate-limiting.
+		defer TestingDisableLargeRowRateLimit()()
 
-		// 1 KiB log threshold scoped to this connection so the row below
-		// (8 KiB primary key) crosses it.
+		// 1 KiB log threshold so the row below (8 KiB primary key) crosses it.
 		db.Exec(t, "SET CLUSTER SETTING sql.guardrails.max_row_size_log = '1KiB'")
 		defer db.Exec(t, "SET CLUSTER SETTING sql.guardrails.max_row_size_log = DEFAULT")
 

